### DISCUSSION TITLE
Rendering a component with slot leads to issues

### DISCUSSION THREAD

**Discussion by djmtype** [formfcw/directus-extension-flexible-editor#47] [Q&A] [answered]
Using Nuxt with Directus. I'm trying to render Tiptap's headings using a custom Heading.vue component. However, I'm running into this warning which I can't figure out the reason for: 
```
Non-function value encountered for default slot. Prefer function slots for better performance.
```
What's needed to fix this?

I've added this to the array of `componentSerializers`:

```js
const componentSerializers: VueComponentSerializers = [
  { 
    name: 'heading', 
    type: 'node', 
    render: (attrs) => 
    [Heading, {...attrs}] 
  },
 ]
```

```js
// Heading.vue
<template>
  <component :is="`h${level}`" :style="headingStyle" class="my-component">
    <slot />
  </component>
</template>

<script setup lang="ts">
import { computed } from 'vue';

const props = defineProps<{
  level: number;
  textAlign?: string;
}>();

const headingStyle = computed(() => {
  return props.textAlign ? { textAlign: props.textAlign } : {};
});
</script>
```

**Comment by djmtype** [comment]
BTW, this is also the case when using the Textfield.vue component that's part of the example.

**Comment by formfcw** [comment, marked as answer]
This has to do with the [tiptap-render-view](https://github.com/formfcw/tiptap-render-view) package and the way the JSON structure is passed to the `renderView` function.

**Reply by formfcw** [reply]
https://stackoverflow.com/questions/69875273/non-function-value-encountered-for-default-slot-in-vue-3-composition-api-comp

**Comment by Krislunde** [comment]
The answer is a bit confusing, is this fixable? What would the implemented fix look like in the code @djmtype provided?

**Reply by formfcw** [reply]
Thanks for your comment. As described above, it is not fixable here, but in the tiptap-render-view package. Feel free to contribute there.
